crates/types/src/traits/signature_key.rs

```diff
@@ -139,3 +139,53 @@ pub trait SignatureKey:
     #[must_use]
     fn genesis_proposer_pk() -> Self;
 }
+
+pub trait BuilderSignatureKey:
```

Comment:
I think we can just use the `SignatureKey` trait for the builder key and don't need a whole new trait.  Is there any reason to create an entirely new trait?

Reply:
https://www.notion.so/espressosys/Builder-fee-payments-4c27ef6c401544359df3ad8a7c59ef08?pvs=4#8ae01e2fe48443dfa8d3d77c82047ffa